### DISCUSSION TITLE
Discarding enqueued jobs per key and/or dropping current jobs

### DISCUSSION THREAD

**Discussion by schrepfler** [jobrunr/jobrunr#1129] [Q&A]
Hi, I'm evaluating JobRun for our use case but from the documentation I don't see if what I have in mind is possible.
In short, I have multiple streams of tasks that need to be executed, let's say groups of tasks for A, and group of tasks for B (there's new groups creating all the time, A and B are essentially unique keys on kafka partitions).
I would be interested in three things:

- Ability to skip any tasks that haven't started for a key and just run the latest and/or 
- ability to cancelling a running task 
- Ability to cancel and/or execute it with the newer arguments.

How much flexibility or control can I retain via the JobRunr API over such execution details?

**Comment by auloin** [comment]
Hi @schrepfler,

You can cancel running tasks by deleting them, and JobRunr will interrupt the task asap (unless it's [not interruptible](https://www.jobrunr.io/en/documentation/background-methods/deleting-jobs/)).

JobRunr Pro allows to easily replace an existing Job, see https://www.jobrunr.io/en/documentation/pro/replace-jobs/.
